snekbox/nsjail.py

```diff
@@ -267,21 +267,32 @@ def python3(
 
             # Parse attachments with time limit
             try:
-                attachments = timed(
-                    MemFS.files_list,
-                    (fs, self.files_limit, self.files_pattern),
-                    {
-                        "preload_dict": True,
-                        "exclude_files": files_written,
-                    },
-                    timeout=self.files_timeout,
-                )
+                with time_limit(self.files_timeout):
```

Comment:
Do we have a test that asserts this time outs?

Reply:
There is a test parsing symlink files for timeout, but that doesn't determine if the timeout occured due to the `time.monotic()` comparison timeout or by the signal. The signal timeout would be if a very large single file was parsed, or pathlib ends up in some loop. 

But also added 2 new tests for `time_limit` successfully interrupting `time.sleep` and a long-running built-in C function (math.factorial) in [1af5bf0](https://github.com/python-discord/snekbox/pull/173/commits/1af5bf0a92a5a65346460590054c1834ff7b7ccc)